lib/common/xml.c

```diff
@@ -1730,6 +1730,17 @@ dump_xml_unformatted(xmlNode * an_xml_node)
     return buffer;
 }
 
+void
+pcmk__xml_serialize_fd_formatted(int fd, xmlNode *cur)
+{
+    xmlOutputBuffer *fd_out = xmlOutputBufferCreateFd(fd, NULL);
+    CRM_ASSERT(fd_out != NULL);
+    xmlNodeDumpOutput(fd_out, cur->doc, cur, 0, 1, NULL);
+    xmlOutputBufferWrite(fd_out, sizeof("\n") - 1, "\n");  /* final NL */
+    xmlOutputBufferClose(fd_out);
```

Comment:
The CI failure is coverity complaining that we're not checking the return value of the above two functions (-1 on error). We could maybe log a warning if either of them returns -1, although we don't have any context about `fd` or `cur` so it couldn't be too helpful. I'm thinking let's just make the new function return EIO on error, and let the callers ignore the return value (callers could log a warning, but it's probably not worth the effort right now).